smartsim/experiment.py

```diff
@@ -155,15 +160,23 @@ def start(self, *jobs: Job) -> tuple[LaunchedJobID, ...]:
         """Execute a collection of `Job` instances.
 
         :param jobs: A collection of other job instances to start
+        :raises TypeError: If jobs provided are not the correct type
+        :raises ValueError: No Jobs were provided.
         :returns: A sequence of ids with order corresponding to the sequence of
             jobs that can be used to query or alter the status of that
             particular execution of the job.
         """
-        # Create the run id
-        run_id = datetime.datetime.now().replace(microsecond=0).isoformat()
-        # Generate the root path
-        root = pathlib.Path(self.exp_path, run_id)
-        return self._dispatch(Generator(root), dispatch.DEFAULT_DISPATCHER, *jobs)
+        if jobs:
+            if not all(isinstance(job, Job) for job in jobs):
+                raise TypeError("jobs argument was not of type Job")
+
+            # Create the run id
+            run_id = datetime.datetime.now().replace(microsecond=0).isoformat()
+            # Generate the root path
+            root = pathlib.Path(self.exp_path, run_id)
+            return self._dispatch(Generator(root), dispatch.DEFAULT_DISPATCHER, *jobs)
+        else:
+            raise ValueError("No jobs provided to start")
```

Comment:
Style nit: Consider inverting these checks just to reduce the amount of indentation
```suggestion
        if not jobs:
            raise ValueError("No jobs provided to start")
        if not all(isinstance(job, Job) for job in jobs):
            raise TypeError("jobs argument was not of type Job")

        # Create the run id
        run_id = datetime.datetime.now().replace(microsecond=0).isoformat()
        # Generate the root path
        root = pathlib.Path(self.exp_path, run_id)
        return self._dispatch(Generator(root), dispatch.DEFAULT_DISPATCHER, *jobs)

```

Reply:
changed